src/main/java/hudson/remoting/URLDeserializationHelper.java

```diff
@@ -69,12 +69,14 @@
     
     private static class SafeURLStreamHandler extends URLStreamHandler {
         @Override
+        @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "Used for safely handling URLs, not for opening a connection.")
```

Comment:
> not for opening a connection

> `return new URL(u.toString()).openConnection(p);`

😕 ?

Reply:
Yeah, this one is kind of iffy. The history of this bit of code is weird. Part of it comes from a specific, odd concern. Looking at this further, it might allow opening URL's on the agent side, if the job requires it. That's part of trusting the server, because the agent jobs can do about anything anyway. I could dig into this one more maybe.